src/compat/posix/fs/dirent/dirent_dvfs.c

```diff
@@ -36,6 +36,12 @@ DIR *opendir(const char *path) {
 		return NULL;
 	}
 
+	if (!S_ISDIR(l.item->flags)) {
```

Comment:
:shipit: 
Could you extract these changes into a separate PR?

Reply:
It was extracted to separate PR.